src/test/java/mate/academy/spring/boot/serviceTests/BookServiceTests.java

```diff
@@ -0,0 +1,163 @@
+package mate.academy.spring.boot.serviceTests;
+
+import mate.academy.spring.boot.dto.book.*;
+import mate.academy.spring.boot.exception.EntityNotFoundException;
+import mate.academy.spring.boot.mapper.BookMapper;
+import mate.academy.spring.boot.model.Book;
+import mate.academy.spring.boot.repository.book.BookRepository;
+import mate.academy.spring.boot.repository.book.BookSpecificationBuilder;
+import mate.academy.spring.boot.service.impl.BookServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTests {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("""
+            Verification of book storage with correct fields""")
+    void testSave_WithCorrectParameters_ShouldReturnSavedBookDto() {
```

Comment:
I recommend you use Given-When-Then construction

Reply:
Please, split your test methods with additional blank line using Given-When-Then construction to make it easier to read